CONTRIBUTING.md

```diff
@@ -47,6 +47,92 @@ Using commit signoffs tags is mandatory for all commits; we also
 encourage that each commit is small and cohesive. See the next sections for details.
 
 
+### Programming style
+
+#### Organization
+
+All code that interacts with the operating system, such as file access, network
+operations, timers, process creation, etc, must happen through a C++
+interface. These interfaces must be organized in a
+`common/<MODULE>/<MODULE>.hpp` hierarchy. The platform implementation must
+reside in `common/<MODULE>/impl/<MODULE>.cpp` files. It is allowed to use more
+files in addition to the `<MODULE>` named files, if necessary.
+
+All the API must be namespaced inside the `<MODULE>` name. Avoid C macros if
+possible, since they can't be namespaced.
+
+#### Code style
+
+Our code style follows [the Google C++ Style
+Guide](https://google.github.io/styleguide/cppguide.html) with a few
+exceptions. To run automatic code style formatting on files you modify, use our
+[clang-format
+template](https://github.com/mendersoftware/mendertesting/blob/master/.clang-format):
+
+```bash
+curl -O https://github.com/mendersoftware/mendertesting/raw/master/.clang-format
+clang-format -i --style=file:.clang-format <MODIFIED_FILES>
+```
+
+Note that the template requires clang-format version 15 or higher.
+
+##### C++ standard
+
+All shared code must use C++ features from no later than the C++11
+standard. Platform code for POSIX platforms is allowed to use C++17 if
+necessary, but C++11 is preferred for consistency with the rest of the code.
+
+#### File names
+
+Files must be named with `.cpp` and `.hpp` extensions. Only use `.h` if the
+header holds C declarations.
+
+##### Line length
+
+Our line lengths are capped at 100 characters instead of 80, which is Google's
+cap. [The other guidelines for line
+length](https://google.github.io/styleguide/cppguide.html#Line_Length) still
+apply.
+
+##### Indentation
+
+Indentation in Mender code uses 1 tab per level, no spaces. Function arguments
+or initializers that are broken over multiple lines are indented one level.
+
+##### Namespaces
+
+All the [rules about
+namespaces](https://google.github.io/styleguide/cppguide.html#Namespaces) apply,
+with one exception: It is allowed to use `using namespace std` to import the
+`std` namespace. All other namespaces must be spelled out where they are used
+(though it is allowed to shorten long names using `namespace short_name =
+long_name`).
```

Comment:
We didn't discuss the second point explicitly (about short names). But I figured this mirrors what is common to do in the Golang world, and makes sense to allow, or?

Reply:
Now that I've seen some of the Boost namespaces, I'm all for this.